packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -20,6 +20,23 @@ import 'platform.dart';
 // ToolExit and a message that is more clear than the FileSystemException by
 // itself.
 
+/// Allow any file system operations executed within the closure to fail with any
+/// operating system error, rethrowing an [Exception] instead of a [ToolExit].
+///
+/// This should not be used with async file system operation.
+///
+/// This can be used to bypass the [ErrorHandlingFileSystem] permission exit
+/// checks for situations where failure is acceptable, such as the flutter
+/// persistent settings cache.
+void withAllowedFailure(void Function() operation) {
```

Comment:
I think this should be a static of `ErrorHandlingFileSystem`. `withAllowedFailure` is a bit too generic by itself. Also, how about `noExitOnFailure` or `exceptionOnFailure` since it's not so much that failures are allowed as it is that they're just presented in a different way.

Reply:
Also, this should account for nesting the calls, and have a test that that is handled correctly.

Reply:
Ahh good point about the nesting, I didn't think of that :)